.env.staging

```diff
@@ -1,6 +1,6 @@
 NEW_EXPENSIFY_URL=https://staging.new.expensify.com/
 SECURE_EXPENSIFY_URL=https://secure.expensify.com/
-EXPENSIFY_URL=https://www.expensify.com/
+EXPENSIFY_URL=https://staging.expensify.com/
```

Comment:
I think we can simplify `.env.{name}` configs by moving the URLs out of  them 

1. Native uses `.env.production` to build the production as well as the staging app 
2. We have a API toggle that tries to toggle between the URL set here (`EXPENSIFY_URL`) and the staging URL. So if we set `EXPENSIFY_URL` to `https://staging.expensify.com/` this would stop working - current logic expects `EXPENSIFY_URL` to be the prod url (e.g. trace how `URL_API_ROOT` and how it's used in `src/libs/HttpUtils.js` 

I propose to declare all the URLs in one place, like you already do in `src/CONST.js` 
Then use environment information to select the correct underlying API, like I try to do in: https://github.com/Expensify/App/pull/15178

This way we also reduce copy pasted code across .env files

Reply:
> I think we can simplify .env.{name} configs by moving the URLs out of them
> I propose to declare all the URLs in one place, like you already do in src/CONST.js
> This way we also reduce copy pasted code across .env files

Yes, I like this idea. Let's do it. Just for visibility we can post this in slack.

> So if we set EXPENSIFY_URL to https://staging.expensify.com/ this would stop working - current logic expects EXPENSIFY_URL to be the prod url (e.g. trace how URL_API_ROOT and how it's used in src/libs/HttpUtils.js

Sorry, I didn't get this. Could you please explain

Reply:
On Staging, there's a Settings -> Preferences -> Test Preferences menu, where you can toggle between the STAGING and PROD APIs, and this depends on `EXPENSIFY_URL` (from .env.staging) to point to the PROD api

See how the code works here
https://github.com/Expensify/App/blob/6971b98076406366dc28e5747321f9d02c6e7753/src/libs/HttpUtils.js#L110-L114

`URL_API_ROOT` is resolved from `EXPENSIFY_URL`

With the change you propose the code would evaluate to something like this:

#### HttpUtil on Web Staging
```js
let apiRoot = 'https://staging.expensify.com/';

 if (shouldUseStagingServer(stagingServerToggleState)) { 
     apiRoot = 'https://staging.expensify.com/';
 } 
```

Right now the code works correctly because `EXPENSIFY_URL` is `https://www.expensify.com/` 
so we have this 

#### HttpUtil on Web Staging
```js
let apiRoot = 'https://www.expensify.com/';

 if (shouldUseStagingServer(stagingServerToggleState)) { 
     apiRoot = 'https://staging.expensify.com/';
 } 
```

Reply:
Ah got it! Thanks for explanation.

Reply:
Thinking more about this. Looks like the two features - 

a). Redirecting to appropriate olddot app and 
b). Making API calls from app to appropriate domain should behave in different ways.

For a). the redirection from newdot to olddot should occur to corresponding environment. i.e. staging to staging, prod to prod and dev to dev.
Whereas in b). the API calls should be made to staging server if "user staging server" setting is toggled on, else should be made to production server.

So, I have reverted the change in `.env.staging` file that was breaking this behavior and is unrelated to this PR. Thanks for catching it. I am +1 to move all this URL logic to a lib and remove them from env files, but I think it should be done in a different PR. We can discuss that and make a new PR to cleanup the logic.